lib/tasks/performance.rake

```diff
@@ -0,0 +1,229 @@
+  task populate_and_simulate_patients: :environment do
+    # Configurable variables
+    target_patients = (ENV['PATIENT_COUNT']|| 500_000).to_i
+    days = (ENV['DAYS'] || 14).to_i
+
+    # Calculated variables
+    num_prototype_patients = [(target_patients * 0.1).to_i, 30_000].min
+
+    ENV['LIMIT'] = num_prototype_patients.to_s
+    # Reduce num_prototype_patients here since demo:populate grows new patient count with each day
+    ENV['COUNT'] = (ENV['COUNT'] || (num_prototype_patients * 0.85) / days).to_i.to_s
+    Rake::Task["demo:populate"].invoke
+
+    ENV['COUNT'] = (target_patients - num_prototype_patients).to_s
+    Rake::Task["demo:create_bulk_data"].invoke
+  end
+
+  desc 'Configure the users in the database for performance testing'
+  task setup_performance_test_users: :environment do
+    raise 'This task is only for use in a development environment' unless Rails.env == 'development' || ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK']
+
+    num_jurisdictions = Jurisdiction.count
+    puts "Creating users for #{num_jurisdictions} jurisdictions\n"
+
+    if !(num_jurisdictions > 50)
+      puts ' Jurisdictions were not found! Make sure to run `PERFORMANCE=true bundle exec rake admin:import_or_update_jurisdictions` or `bundle exec rails perf:populate`'
+      exit(1)
+    end
+
+    num_users = User.count
+    unless num_users.zero?
+      puts 'This task should only be run when no users exist!'
+      puts "There are currently #{num_users} users."
+      exit(1)
+    end
+    users = []
+
+    # Super User at the USA level
+    usa = Jurisdiction.find_by_name('USA')
+    usa_user = User.create!(
+      email: "#{usa.unique_identifier}_super_user@example.com",
+      password: '1234567ab!',
+      role: Roles::SUPER_USER,
+      jurisdiction_id: usa.id,
+      force_password_change: false,
+      authy_enabled: false,
+      authy_enforced: false
+    )
+
+    prototype_user = {
+      email: usa_user[:email],
+      encrypted_password: usa_user[:encrypted_password],
+      sign_in_count: usa_user[:sign_in_count],
+      current_sign_in_at: usa_user[:current_sign_in_at],
+      last_sign_in_at: usa_user[:last_sign_in_at],
+      current_sign_in_ip: usa_user[:current_sign_in_ip],
+      last_sign_in_ip: usa_user[:last_sign_in_ip],
+      failed_attempts: usa_user[:failed_attempts],
+      locked_at: usa_user[:locked_at],
+      force_password_change: usa_user[:force_password_change],
+      jurisdiction_id: usa_user[:jurisdiction_id],
+      password_changed_at: usa_user[:password_changed_at],
+      created_at: usa_user[:created_at],
+      updated_at: usa_user[:updated_at],
+      authy_id: usa_user[:authy_id],
+      last_sign_in_with_authy: usa_user[:last_sign_in_with_authy],
+      authy_enabled: usa_user[:authy_enabled],
+      authy_enforced: usa_user[:authy_enforced],
+      api_enabled: usa_user[:api_enabled],
+      role: usa_user[:role],
+      is_api_proxy: usa_user[:is_api_proxy]
```

Comment:
do most of these metadata fields even need to be specified? since most of them aren't defined for `usa_user`, wouldn't they just be `nil` or a default value (if there is one) and when the rest of the users are written to db, wouldn't these fields also be `nil` or the default value?